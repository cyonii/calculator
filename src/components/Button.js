import PropTypes from 'prop-types';

const Button = (props) => {
  const { name, clickHandler } = props;
  let { className } = props;
  className += ' calc-btn w-100 border rounded-pill fs-3 fw-bold py-2';

  const handleClick = () => clickHandler(name);

  return (
    <button
      type="button"
      className={`btn ${className}`}
      onClick={handleClick}
      data-testid={`btn-${name}`}
    >
      {name}
    </button>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  className: 'btn-light',
};

export default Button;
