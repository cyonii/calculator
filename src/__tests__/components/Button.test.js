import { render, screen, cleanup } from '@testing-library/react';
import Button from '../../components/Button';

describe('<Button /> Component', () => {
  let testButton;

  beforeEach(async () => {
    testButton = <Button name="1" clickHandler={jest.fn} />;
  });

  afterEach(() => cleanup);

  it('should render correctly', () => {
    render(testButton);

    const buttonElement = screen.getByTestId('btn-1');
    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent('1');
  });

  test('should match snapshot', () => {
    expect(testButton).toMatchSnapshot();
  });
});
