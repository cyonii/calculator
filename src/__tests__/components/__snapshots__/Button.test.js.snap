// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component should match snapshot 1`] = `
<Button
  className="btn-light"
  clickHandler={[Function]}
  name="1"
/>
`;
